### PR TITLE
feat: transparency api consumed for dashboard page and latest donatio…

### PR DESCRIPTION
**Pull request by RB-Isiaq**
> Transparency dashboard api integrated.

Screenshot 2024-06-13 185129
Screenshot 2024-06-13 185041
Screenshot 2024-06-13 185013
Screenshot 2024-06-13 184951
Screenshot 2024-06-13 184929
Screenshot (70)

https://github.com/edustipend/dotorg/assets/99425435/88833777-a993-40a9-911e-53d98fcbd5b8

### CODE REVIEW

File: src/sections/TransparencyDashboard/constants.js [hunk 1]
```diff
@@ -136,6 +136,17 @@ export const data = {
   ]
 };
 
+export const resData = {
+  goal: 10000000,
```
Comment (edustipend-root) [inline src/sections/TransparencyDashboard/constants.js:140]: @RB-Isiaq, the goal is 11,000,000.

**Comment by edustipend-root** [comment]
@RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?

File: src/sections/SupportALearner/LatestDonations/LatestDonations.jsx [hunk 2]
```diff
@@ -46,7 +72,7 @@ const LatestDonations = () => {
             </div>
           ))}
         </div>
-        <Pagination currentPage={currentPage} onPageChange={changePage} showViewAll={true} totalPages={3} />
+        <Pagination currentPage={currentPage} onPageChange={changePage} showViewAll={true} totalPages={total} handleNextCall={() => {}} />
```
Comment (edustipend-root) [inline src/sections/SupportALearner/LatestDonations/LatestDonations.jsx:75]: `handleNextCall` should route to the Donations dashboard page.

**Comment by RB-Isiaq** [comment]
> @RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?

It has been fixed sir @edustipend-root

**Comment by edustipend-root** [comment]
> > @RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?
> 
> It has been fixed sir @edustipend-root

@RB-Isiaq, here's the screenshot of what I'm seeing when I go to https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/transparency-dashboard

image

So couple of things here:

- Why are we seeing NaN values?
- If you look at the screenshot closely, you'd see that this build is calling PROD api instead of DEV? Can you please check why that is happening? This should NOT be calling PROD api directly, it should be calling the DEV instance of the service.

Please investigate this urgently. I'm really hoping we have not been making any calls to our PROD api.

**Review by edustipend-root — CHANGES_REQUESTED** [review]
See my comment

**Comment by edustipend-root** [comment]
Also @RB-Isiaq, can we change the route for the dashboard to `/support-a-learner/transparency-dashboard` ?

**Comment by RB-Isiaq** [comment]
> > > @RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?
> > 
> > 
> > It has been fixed sir @edustipend-root
> 
> @RB-Isiaq, here's the screenshot of what I'm seeing when I go to https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/transparency-dashboard
> 
> image
> 
> So couple of things here:
> 
> * Why are we seeing NaN values?
> * If you look at the screenshot closely, you'd see that this build is calling PROD api instead of DEV? Can you please check why that is happening? This should NOT be calling PROD api directly, it should be calling the DEV instance of the service.
> 
> Please investigate this urgently. I'm really hoping we have not been making any calls to our PROD api.

So @edustipend-root  about this, the pr links actually call the prod api all the time, probably because it sees it as a production site.
This is a screenshot from @Ezeh20 pr.
Screenshot (76)
Screenshot (75)

image
Screenshot (77)

But then, when you run it up on localhost, it sees it as the dev environment, and uses the dev api.
See this sir:
image
image

**Comment by edustipend-root** [comment]
> > > > @RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?
> > > 
> > > 
> > > It has been fixed sir @edustipend-root
> > 
> > 
> > @RB-Isiaq, here's the screenshot of what I'm seeing when I go to https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/transparency-dashboard
> > image
> > So couple of things here:
> > 
> > * Why are we seeing NaN values?
> > * If you look at the screenshot closely, you'd see that this build is calling PROD api instead of DEV? Can you please check why that is happening? This should NOT be calling PROD api directly, it should be calling the DEV instance of the service.
> > 
> > Please investigate this urgently. I'm really hoping we have not been making any calls to our PROD api.
> 
> So @edustipend-root about this, the pr links actually call the prod api all the time, probably because it sees it as a production site. This is a screenshot from @Ezeh20 pr. Screenshot (76) Screenshot (75)
> 
> image Screenshot (77)
> 
> But then, when you run it up on localhost, it sees it as the dev environment, and uses the dev api. See this sir: image image

Thanks @RB-Isiaq for investigating this, this is gonna be a problem because when we want the team to test tomorrow, we want them testing against DEV api not PROD api. I have to figure out a way for the build to point to DEV not PROD.

I'll approve and get this merged in for now.

**Review by edustipend-root — APPROVED** [review]

**Comment by RB-Isiaq** [comment]
> > > > > @RB-Isiaq, I'm seeing an error when I visit https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/support-a-learner . Can you please take a look?
> > > > 
> > > > 
> > > > It has been fixed sir @edustipend-root
> > > 
> > > 
> > > @RB-Isiaq, here's the screenshot of what I'm seeing when I go to https://edustipenddotorg--pr174-feat-transparency-ap-tectirtx.web.app/transparency-dashboard
> > > image
> > > So couple of things here:
> > > 
> > > * Why are we seeing NaN values?
> > > * If you look at the screenshot closely, you'd see that this build is calling PROD api instead of DEV? Can you please check why that is happening? This should NOT be calling PROD api directly, it should be calling the DEV instance of the service.
> > > 
> > > Please investigate this urgently. I'm really hoping we have not been making any calls to our PROD api.
> > 
> > 
> > So @edustipend-root about this, the pr links actually call the prod api all the time, probably because it sees it as a production site. This is a screenshot from @Ezeh20 pr. Screenshot (76) Screenshot (75)
> > image Screenshot (77)
> > But then, when you run it up on localhost, it sees it as the dev environment, and uses the dev api. See this sir: image image
> 
> Thanks @RB-Isiaq for investigating this, this is gonna be a problem because when we want the team to test tomorrow, we want them testing against DEV api not PROD api. I have to figure out a way for the build to point to DEV not PROD.
> 
> I'll approve and get this merged in for now.

Alright no problem, Thank you. @edustipend-root

**Pull request merged**